<!--
Example launch file: launches the scan matcher with pre-recorded data
-->

<launch>
    <arg name="IS_TWISTSTAMPED" default="true" />
    <arg name="publish_covariance" default="false"/>

    <param name="/stamped_vel" value="$(arg IS_TWISTSTAMPED)"/>

    <node pkg="daart" type="scan_copy" name="scan_copy" />

    <node pkg="daart" type="daart_odom_clean_node" name="daart_odom_clean_node">
        <param name="topic2clean" value="odom_encoder" />
        <param name="topicPub" value="odom_encoder2" />
        <param name="childFrame" value="base_link_encoder2" />
        <param name="frameId" value="odom_encoder2" />
    </node>

    #### start the laser scan_matcher ##############################
    <group if="$(arg publish_covariance)">
        <param name="laser_scan_matcher_node/do_compute_covariance" value="1"/>
        <param name="laser_scan_matcher_node/publish_pose_with_covariance" value="true"/>
        <param name="laser_scan_matcher_node/publish_pose_with_covariance_stamped" value="true"/>
    </group>

    <node pkg="laser_scan_matcher" type="laser_scan_matcher_node" name="laser_scan_matcher_node" output="screen">
        <remap from="scan" to="scan_copy2odom"/>
        <remap from="odom" to="odom_encoder2"/>
        <param name="max_iterations" value="10"/>
        <param name="base_frame" value="base_link_scan"/>
        <param name="fixed_frame" value="odom_scan"/>
        <param name="use_odom" value="true"/>
        <param name="publish_pose_stamped" value="true"/>
        <param name="use_imu" value="false"/>
    </node>

    <!-- <node pkg="daart" type="daart_odom_scan_node" name="daart_odom_scan_node" output="screen" >
        <param name="robotX" value="7.0" />
        <param name="robotY" value="5" />
        <param name="robotTh" value="0" />
    </node> -->

    <node pkg="daart" type="pose2odom" name="pose2odom" output="screen" >
        <param name="robotX" value="7.0" />
        <param name="robotY" value="5" />
        <param name="robotTh" value="0" />
<!--        <param name="child_frame_id" value="base_link" />
        <param name="frame_id" value="odom" />-->
    </node>

    <node pkg="tf" type="static_transform_publisher" name="lasercopy2odom2odomscan" args="0 0 0 0 0 0  base_link_scan laser_copy2odom 1000" />

</launch>
